src/controller/create.controller.ts

```diff
@@ -0,0 +1,199 @@
+// tslint:disable:no-shadowed-variable
+import { CreateView as CreateViewDefault } from '../view/create.view';
+import {
+  DefaultStructureService,
+  IFunctionConfig,
+} from '../service/defaultStructure.service';
+import { LoginController } from './login.controller';
+import { factory } from '../service/faasFactory.service';
+
+interface IScheduleConfig {
+  lambdaUUID?: string;
+  cronExpression?: string;
+  isActive?: boolean;
+}
+
+interface ICreateControllerConfig {
+  createView?: CreateViewDefault;
+  defaultStructureService?: DefaultStructureService;
+  loginController?: LoginController;
+  uuid?: string;
+}
+
+export class CreateController {
+  private readonly createView: CreateViewDefault;
+
+  private readonly defaultStructureService: DefaultStructureService;
+
+  private loginController: LoginController;
+
+  constructor(
+    /* istanbul ignore next */ {
+      createView = new CreateViewDefault(),
+      defaultStructureService = new DefaultStructureService(),
+      loginController = new LoginController(),
+    }: ICreateControllerConfig = {},
+  ) {
+    this.createView = createView;
+    this.defaultStructureService = defaultStructureService;
+    this.loginController = loginController;
+  }
+
+  /**
+   *
+   * @param {IFunctionCreateOptions} - Passed function name and flags
+   * @returns {Promise<void>} - create view
+   * @memberof CreateController
+   */
+  public async createFunction(functionParameters = {}): Promise<void> {
+    try {
+      const missingParameters = await this.checkAndAskForMissingFunctionParameters(
+        functionParameters,
+      );
+
+      const functionConfig: IFunctionConfig = {
+        ...functionParameters,
+        ...missingParameters,
+      };
+
+      this.defaultStructureService.createFunction(functionConfig);
+      this.createView.showFunctionIsCreated(functionConfig);
+    } catch (error) {
+      this.createView.showErrorMessage(error.message);
+    }
+  }
+
+  private async checkAndAskForMissingFunctionParameters(
+    incompleteFunctionParameters: IFunctionConfig = {},
+  ): Promise<any> {
+    const { name, description, event } = incompleteFunctionParameters;
+
+    let selectedFunctionParameters = {};
+
+    if (!name) {
+      selectedFunctionParameters = {
+        ...selectedFunctionParameters,
+        ...(await this.createView.askForFunctionName()),
+      };
+    }
+
+    if (!description) {
+      selectedFunctionParameters = {
+        ...selectedFunctionParameters,
+        ...(await this.createView.askForFunctionDescription()),
+      };
+    }
+
+    if (!event) {
+      let selectedEventId;
+
+      if (await this.isUserLoggedIn()) {
```

Comment:
`await this.isUserLoggedIn()` <-- Don't we need the user to be logged in for creating a function as well?

If that's the case we can call it over there. And then this sections becomes a bit cleaner

Reply:
not necessarily. The lambda creation works offline. You push a lambda from your local environment to the server. This allows you to select an event from a list of events (pulled from the backend when logged in). Otherwise you have to type it in manually.